lm/vocabulary.py

```diff
@@ -146,3 +146,47 @@ def run(self):
         with open(self.out_vocabulary.get_path(), "w") as fout:
             for word in sorted(vocabulary):
                 fout.write(f"{word}\n")
+
+
+class VocabularyFromTextJob(Job):
+    """
+    Extract vocabulary from given text files based on frequency.
+    """
+
+    def __init__(self, file_paths, size=1000000, counts=False):
+        """
+        :param List[Path] file_paths: paths to the text files
+        :param int size: expected size of the vocabulary
+        :param bool counts: whether write the counts of the words into the vocabulary file
```

Comment:
I find `counts` as a flag to activate behaviour not clear. maybe better `add_counts` or `include_counts`.

Or what it actually does is select the output format so one could do an enum and select supported formats based on this (in case we want to have a different output format in the future it is easier to extend).

Reply:
Tying into this discussion: Different classes in this File have different input formats (`FromLexicon`, `FromLm`, and now `FromText`) and I understand we need different parsers in each case. But there are also different output formats in each case:
* Lexicon: docstring says `"{word}: {index}\n"` but implementation seems `"{word} {index}\n"`
* Lm: only `"{word}\n"`
* Text: both `"{word}\n"` and `"{word} {count}\n"`

This is very inconsistent. Should we aim to consistencise it? 
What are the necessary vocab file formats across srilm, kenlm, subwordnmt, returnn, pytorch, ... ?

Reply:
For SRILM and KenLM, we need `"{word}\n"`
For RETURNN, we need `"{word} {index}\n"`
The new job will output `"{word}\n"` by default, and it could also output `"{word} {count}\n"` if the user want, e.g. if the user want to check the counts or merge different vocabularies.

Reply:
For RETURNN: what's `index`? 0, 1, ...? If so, then we might be able to use another job to obtain such an index (or even a loop enclosed in `enumerate`).

I find `f"{word} {count}\n"` to be the most consistent output, but I'm afraid that changing current functionality breaks some developed jobs.

Reply:
> For RETURNN: what's index? 0, 1, ...? 

yes, but we usually map sentence begin and sentence end to the same token. So it will be 0, 0, 1, 2, ... (depending on position of `<s>` and `<\s>`)